### PR TITLE
Feature/merge dataframe into table

### PR DESCRIPTION
**Pull request by LauJohansson**
Two methods:

merge_df_into_target
and
concat_dfs

### CODE REVIEW

File: src/atc/transformations.py [hunk 1]
```diff
@@ -197,3 +204,168 @@ def join_time_series_dataframes(
...
+        {', '.join(f"target.{col} = source.{col}" for col in non_join_cols)}
+        WHEN NOT MATCHED
+        THEN INSERT -- insert new records
+        (
+        {', '.join(df.columns)}
+        )
+        VALUES
+        (
+        {', '.join(f"source.{col}" for col in df.columns)}
+        )
```
Comment (mrmasterplan) [inline src/atc/transformations.py:315]: `WHEN MATCHED THEN UPDATE SET * WHEN NOT MATCHED THEN INSERT *` is much more concise and does the same. See the spark documentation.

File: src/atc/transformations.py [hunk 2]
```diff
@@ -197,3 +204,168 @@ def join_time_series_dataframes(
...
+    all_cols = sorted(list(set(all_cols)))
+
+    # Create a list to save missing columns for each dataframe
+    missing = []
+
+    # For each dataframe: Get missing columns and add them to the dataframe
+    for i, df in enumerate(dfs):
+        missing.append([col for col in all_cols if col not in df.columns])
+
+        for col in missing[i]:
```
Comment (mrmasterplan) [inline src/atc/transformations.py:358]: - `missing` is never used outside the for loop. there is no reason to keep this state for later.
- the list comprehension that you construct adds nothing here.
- recommendation: do a simple set of nested for loops
```
    for i, df in enumerate(dfs):
         for col in all_cols:
            if col not in df.columns: 
                dfs_updated[i] = dfs_updated[i].withColumn(col, F.lit(None))
```
- this code is much cleaner and would have been easier for me to read.

File: tests/functions/test_drop_table_cascade.py [hunk 3]
```diff
@@ -0,0 +1,9 @@
+import unittest
+
+@unittest.skip("No tests has yet been implemented.")
+# This file should test the function "drop_table_cascade" in atc/functions
```
Comment (mrmasterplan) [inline tests/functions/test_drop_table_cascade.py:4]: maybe add a comment that our current test pipeline does not support delta tables yet.
Also in the other skipped test.

Comment (LauJohansson) [reply tests/functions/test_drop_table_cascade.py:4]: I have rewritten the comment (y).

**Pull request merged**